### PR TITLE
Fix double initialisation of variables in ASR

### PR DESCRIPTION
**Pull request by czgdp1807**
Closes https://github.com/lcompilers/lpython/issues/1626

### CODE REVIEW

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -2210,24 +2237,30 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 current_procedure_abi_type, s_access, s_presence,
                 value_attr);
         ASR::symbol_t* v_sym = ASR::down_cast<ASR::symbol_t>(v);
+        ASR::Variable_t* v_variable = ASR::down_cast<ASR::Variable_t>(v_sym);
 
-        if( init_expr && current_body) {
+        if( init_expr && current_body &&
+            (is_runtime_expression || !is_variable_const)) {
```
Comment (czgdp1807) [inline src/lpython/semantics/python_ast_to_asr.cpp:2243]: Only those variables will be initialised by converting to assignments which are non-constant or have a runtime expressions. Also an error is thrown if a constant variable is not initialised at declaration (see lines 2215 to 2219).

File: src/libasr/asr_verify.cpp [hunk 2]
```diff
@@ -516,6 +516,15 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
             "Variable::m_parent_symtab must be present in the ASR ("
                 + std::string(x.m_name) + ")");
 
+        if( symtab->parent == nullptr ) {
+            // For nowRestrict this check only to variables which are present
```
Comment (certik) [inline src/libasr/asr_verify.cpp:520]:
```suggestion
            // For now restrict this check only to variables which are present
```

**Review by certik — APPROVED** [review]
I think that this is good, thanks for fixing it and adding a test!

I suggest we merge it, and if there are more issues, we'll tackle it as we discover it.

**Comment by czgdp1807** [comment]
Okay. Let me check if verify changes work with LFortran. Then I will merge both the PRs.

**Review by certik — APPROVED** [review]

**Pull request merged**